### ISSUE TITLE
Simple method to check connection to Neo4j database

### ISSUE THREAD

**Issue by wgevaert** [neo4j-php/neo4j-php-client#90]
# Use case
I am trying to build a website that allows the user to write data to a Neo4j-database via a POST.
Before they try to write data to the database (i.e. in the GET that gives them their form), I would like to check that the Neo4j-database is available, and show a warning when this is not the case.

# Feature request
A simple function in (I think) `Contracts\DriverInterface` or `Contracts\SessionInterface` such as `hasSession() : bool` or `canWriteToDatabase() : bool` that I can use.

# Considered alternatives
I thought of using `Contracts\DriverInterface::createSession()` and when it throws an error this can be interpreted as the database having problems, but I am not confident that this is the best way to do this, and the docBlock of this function does not mention anything about throwing exceptions.

Any suggestions are welcome!

**Comment by transistive** [comment]
Hi @wgevaert,

Thank you for taking the time to request the feature! The use case is very interesting. All connections are lazily opened, meaning they will only throw an error once you want to run queries on them:

```php
\Laudis\Neo4j\ClientBuilder::create()
    ->withDriver('default', 'bolt://localboast') // connection is only configured here, but not yet opened
    ->build() // still no connection being made
    ->getDriver('default') // still no connection being made
    ->createSession() // still no connection being made
    ->run('RETURN 1 AS one'); // errors here
```
The current way would be to simply run and check for a connection exception. But when looking at other drivers in different languages: they seem to provide a `verifyConnectivity()` method on the driver. https://github.com/neo4j/neo4j-java-driver/blob/ef1bce7da9cc54a0a5b8c1c735114ae82030b38f/driver/src/main/java/org/neo4j/driver/Driver.java#L182 I would personally return a boolean instead of an exception in this case, so I would call it `canMakeValidConnection()` and return a boolean. We can then also provide a shorthand method on the `Client` object. 

What do you think?

Kind regards,

Ghlen

**Comment by fbiville** [comment]
To better align with official, I'd kindly suggest to name the method `verifyConnectivity` (whether it throws or returns a boolean is left as an idiomatic choice, I'd say).

**Comment by transistive** [comment]
No problem! I changed the method name since I did not release it yet. I will probably release it tomorrow if it is green lit :+1:

**Comment by wgevaert** [comment]
> if it is green lit

Am I supposed to say that the changes look great or should someone else do that?
I think it looks great!

**Comment by transistive** [comment]
Perfect! It has now been released on version 2.3.0!

Good luck!
